acceptance/suites/tests/030-tk-auth/010-tk-auth-smoke.rb

```diff
@@ -0,0 +1,146 @@
+confine :except, :platform => 'windows'
+
+test_name "TK Auth Smoke Test Presuite" do
+  
+  #TODO: skip if pe
+  step 'Configure the server setting' do
+    # so that we don't have to say puppet agent -t --server={fqdn}
+    master_fqdn=on(master, 'facter fqdn').stdout.chomp
+    hosts.each do |h|
+      on(h, "puppet config set server #{master_fqdn}") 
+    end
+  end
+
+  step 'Install required modules available on puppet forge' do
+    on(master, 'puppet resource package git ensure=present')
+    on(master, 'puppet module install puppetlabs-vcsrepo')
+    on(master, 'puppet module install puppetlabs-stdlib')
+    on(master, 'puppet module install puppetlabs-concat')
+    on(master, 'puppet module install puppetlabs-hocon')
+  end
+
+  #TODO: skip if present 
+  step 'Acquire a git auth key ' do
+    gitsshkey=File.file?("#{ENV['HOME']}/.ssh/id_rsa") ? "#{ENV['HOME']}/.ssh/id_rsa" : File.expand_path('~/.ssh/id_rsa-jenkins')
+    master.do_scp_to(gitsshkey, '/root/.ssh/gitkey',options)
+    on(master, 'eval $(ssh-agent)')
+  end
+
+  #TODO: skip if present
+  step 'Install hocon gem' do
+    #TODO: Is there a way we can calculate this location from puppet config?
+    on(master, '/opt/puppetlabs/puppet/bin/gem install hocon')
+  end
+
+#TODO: Make this look in forge first, and then proceed to git if forge fails.
+  step 'Install puppetlabs-hocon and puppetlabs-http_authorization modules from git' do
+    environmentpath=on(master, 'puppet config print environmentpath').stdout.chomp
+    environment='production'
+    $manifest_path=[environmentpath,environment,'manifests','site.pp'].join('/')
```

Comment:
Wasn't sure why this is named with a `$` at the start?  Also, maybe could use Ruby's `File.join` to put the full path together here - http://ruby-doc.org/core-1.9.3/File.html#method-c-join?